drivers/wifi/nxp/nxp_wifi_drv.c

```diff
@@ -0,0 +1,1829 @@
+		memcpy(ap_sta_info.mac, disassoc_resp->sta_addr, WIFI_MAC_ADDR_LEN);
+		wifi_mgmt_raise_ap_sta_disconnected_event(g_uap.netif, &ap_sta_info);
+
+		LOG_DBG("WLAN: UAP a Client Dissociated:");
+		LOG_DBG(" Client MAC => ");
+		print_mac((const char *)(disassoc_resp->sta_addr));
+		LOG_DBG(" Reason code => ");
+		LOG_DBG("%d", disassoc_resp->reason_code);
+		break;
+	case WLAN_REASON_UAP_STOPPED:
+		net_eth_carrier_off(g_uap.netif);
+		LOG_DBG("WLAN: UAP Stopped");
+
+		net_dhcpv4_server_stop(g_uap.netif);
+		LOG_DBG("DHCP Server stopped successfully");
+		s_nxp_wifi_UapActivated = false;
+		break;
+#endif
+	case WLAN_REASON_PS_ENTER:
+		LOG_DBG("WLAN: PS_ENTER");
+		break;
+	case WLAN_REASON_PS_EXIT:
+		LOG_DBG("WLAN: PS EXIT");
+		break;
+#ifdef CONFIG_NXP_WIFI_SUBSCRIBE_EVENT_SUPPORT
+	case WLAN_REASON_RSSI_HIGH:
+	case WLAN_REASON_SNR_LOW:
+	case WLAN_REASON_SNR_HIGH:
+	case WLAN_REASON_MAX_FAIL:
+	case WLAN_REASON_BEACON_MISSED:
+	case WLAN_REASON_DATA_RSSI_LOW:
+	case WLAN_REASON_DATA_RSSI_HIGH:
+	case WLAN_REASON_DATA_SNR_LOW:
+	case WLAN_REASON_DATA_SNR_HIGH:
+	case WLAN_REASON_LINK_QUALITY:
+	case WLAN_REASON_PRE_BEACON_LOST:
+		break;
+#endif
+	default:
+		LOG_WRN("WLAN: Unknown Event: %d", reason);
+	}
+	return 0;
+}
+
+static int nxp_wifi_wlan_init(void)
+{
+	int status = NXP_WIFI_RET_SUCCESS;
+	int ret;
+
+	if (s_nxp_wifi_State != NXP_WIFI_NOT_INITIALIZED) {
+		status = NXP_WIFI_RET_FAIL;
+	}
+
+	if (status == NXP_WIFI_RET_SUCCESS) {
+		k_event_init(&s_nxp_wifi_SyncEvent);
+	}
+
+	if (status == NXP_WIFI_RET_SUCCESS) {
+		ret = wlan_init(wlan_fw_bin, wlan_fw_bin_len);
+		if (ret != WM_SUCCESS) {
+			status = NXP_WIFI_RET_FAIL;
+		}
+	}
+
+	if (status == NXP_WIFI_RET_SUCCESS) {
+		s_nxp_wifi_State = NXP_WIFI_INITIALIZED;
+	}
+
+	if (status != NXP_WIFI_RET_SUCCESS) {
+		return -1;
+	}
+
+	nxp_wifi_internal_register_rx_cb(nxp_wifi_recv);
+
+	return 0;
+}
+
+static int nxp_wifi_wlan_start(void)
+{
+	int status = NXP_WIFI_RET_SUCCESS;
+	int ret;
+	uint32_t syncBit;
+
+	if (s_nxp_wifi_State != NXP_WIFI_INITIALIZED) {
+		status = NXP_WIFI_RET_NOT_READY;
+	}
+
+	if (status == NXP_WIFI_RET_SUCCESS) {
+		k_event_clear(&s_nxp_wifi_SyncEvent, NXP_WIFI_SYNC_INIT_GROUP);
+
+		ret = wlan_start(nxp_wifi_wlan_event_callback);
+		if (ret != WM_SUCCESS) {
+			status = NXP_WIFI_RET_FAIL;
+		}
+	}
+
+	if (status == NXP_WIFI_RET_SUCCESS) {
+		syncBit = k_event_wait(&s_nxp_wifi_SyncEvent, NXP_WIFI_SYNC_INIT_GROUP, false,
+				       NXP_WIFI_SYNC_TIMEOUT_MS);
+		k_event_clear(&s_nxp_wifi_SyncEvent, NXP_WIFI_SYNC_INIT_GROUP);
+		if (syncBit & NXP_WIFI_EVENT_BIT(WLAN_REASON_INITIALIZED)) {
+			status = NXP_WIFI_RET_SUCCESS;
+		} else if (syncBit & NXP_WIFI_EVENT_BIT(WLAN_REASON_INITIALIZATION_FAILED)) {
+			status = NXP_WIFI_RET_FAIL;
+		} else {
+			status = NXP_WIFI_RET_TIMEOUT;
+		}
+	}
+
+	if (status == NXP_WIFI_RET_SUCCESS) {
+		s_nxp_wifi_State = NXP_WIFI_STARTED;
+	}
+
+	if (status != NXP_WIFI_RET_SUCCESS) {
+		return -1;
+	}
```

Comment:
This could be done other way
```
if (status != NXP_WIFI_RET_SUCCESS) {
		return -1;
}

s_nxp_wifi_State = NXP_WIFI_STARTED;
```
This way one `if` check can be omitted

Reply:
thanks for your advice, updated